### DISCUSSION TITLE
Gatsby not working on Apple M1

### DISCUSSION THREAD

**Discussion by swve** [gatsbyjs/gatsby#29854] [Help] [answered]
## Description

I can't use Gatsby in a arm64 environment, i'm using a Mac under Apple M1 processor 

### Steps to reproduce

- npm install 

### Expected result

Project installation 

### Actual result

Errors and warning about packages that are not available yet for an arm environment 

### Environment

  System:
    OS: macOS 11.2.2
    CPU: (8) arm64 Apple M1
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.8.0 - /opt/homebrew/bin/node
    npm: 7.5.0 - /opt/homebrew/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.192
    Firefox: 86.0
    Safari: 14.0.3
  npmGlobalPackages:
    gatsby-cli: 2.19.1

**Comment by LekoArts** [comment]
That's a problem not specific to Gatsby. You'll need to check for yourself how you can use the necessary dependencies (Node, sharp, etc.) with this new environment. If you Google, you'll find resources like:

https://www.youtube.com/watch?v=MI-Q4DyQWak
https://github.com/lovell/sharp/issues/2460

**Comment by Cecile-Lebleu** [comment]
I was able to fix that by running the following: (one line at a time)

```
brew info vips
brew reinstall vips
xcode-select --install
brew install gcc
brew reinstall vips
brew info vips
npm i
```

**Comment by LekoArts** [comment, marked as answer]
sharp v0.28 now provides M1 support so manual installation of vips won't be necessary anymore: https://github.com/lovell/sharp/issues/2460#issuecomment-809499444
